### DISCUSSION TITLE
Problem with .erlang.cookie after upgrade from 3.8.21 to 3.9.5

### DISCUSSION THREAD

**Discussion by bmaehr** [rabbitmq/rabbitmq-server#3347] [Other]
Hello,

I tried to upgrade rabbitmq running on k8s from 3.8.21 to 3.9.5 and get the common "Cookie file /var/lib/rabbitmq/.erlang.cookie must be accessible by owner only" error. But the permission is set correct.

```
bash-5.1$ ls -a -l
total 36
drwxrwsr-x    4 root     rabbitmq      4096 Aug 30 21:20 .
drwxr-xr-x    1 root     root          4096 Feb  2  2021 ..
-rw-------    1 rabbitmq rabbitmq       462 Aug 30 21:08 .bash_history
-r--------    1 rabbitmq rabbitmq        20 Aug 30 00:00 .erlang.cookie
drwxrws---    2 root     rabbitmq     16384 Feb  9  2021 lost+found
drwxrwsr-x    4 rabbitmq rabbitmq      4096 Aug 30 20:09 mnesia
bash-5.1$
```

```
error: {
  badmatch, {
    error, { {
        shutdown, {
          failed_to_start_child,
          auth, {
            "Cookie file /var/lib/rabbitmq/.erlang.cookie must be accessible by owner only",
            [{
                auth,
                init_cookie,
                0,
                [{
                    file,
                    "auth.erl"
                  }, {
                    line,
                    290
                  }
                ]
              }, {
                auth,
                init,
                1,
                [{
                    file,
                    "auth.erl"
                  }, {
                    line,
                    144
                  }
                ]
              }, {
                gen_server,
                init_it,
                2,
                [{
                    file,
                    "gen_server.erl"
                  }, {
                    line,
                    423
                  }
                ]
              }, {
                gen_server,
                init_it,
                6,
                [{
                    file,
                    "gen_server.erl"
                  }, {
                    line,
                    390
                  }
                ]
              }, {
                proc_lib,
                init_p_do_apply,
                3,
                [{
                    file,
                    "proc_lib.erl"
                  }, {
                    line,
                    226
                  }
                ]
              }
            ]
          }
        }
      }, {
        child,
        undefined,
        net_sup_dynamic, {
          erl_distribution,
          start_link,
          [[rabbit_prelaunch_11 @ localhost, shortnames], false, net_sup_dynamic]
        },
        permanent,
        false,
        1000,
        supervisor,
        [erl_distribution]
      }
    }
  }
}
```

Is the (old) solution from https://github.com/docker-library/rabbitmq/issues/171 to map the folder `/var/lib/rabbitmq/mnesia` instead of `/var/lib/rabbitmq` the correct solution? Why was it not necessary on 3.8.21?

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
This message is [from Erlang/OTP](https://github.com/erlang/otp/blob/107af3c100b2e0dc413e51cde230ddd102db7b7e/lib/kernel/src/auth.erl#L404) and my guess is that it's coming from a CLI tools call. CLI tools can use a different cookie file path.

In `3.9.5`, `RABBITMQ_ERLANG_COOKIE` is supported by [RabbitMQ itself](https://github.com/docker-library/rabbitmq/issues/513) and is no longer used by the Docker image. The value must be the value of the cookie, not a file path. That may be another possible reason for confusion.

**Reply by bmaehr** [reply]
But if the path `/var/lib/rabbitmq/.erlang.cookie` is written to the error message I would expect that thats's really the path the process is looking at. 

Could it be that the process is running as an other user than `rabbitmq`?
Or could it be that process get's wrong information about the permission because of the volume mount?

**Comment by michaelklishin** [comment]
`strace` is the ultimate way to find out what processes try to access the file (or a different path) when loading the cookie. The 3.9 version of the Docker image has [dropped support for most environment variables](https://github.com/docker-library/rabbitmq/pull/467) to become drastically simpler.
